pkg/agent/openflow/client.go

```diff
@@ -825,11 +825,12 @@ func (c *client) InstallExternalFlows(exceptCIDRs []net.IPNet) error {
 			ipv4CIDRs = append(ipv4CIDRs, cidr)
 		}
 	}
-	if c.nodeConfig.NodeIPv4Addr != nil && c.nodeConfig.PodIPv4CIDR != nil {
-		flows = c.externalFlows(c.nodeConfig.NodeIPv4Addr.IP, *c.nodeConfig.PodIPv4CIDR, localGatewayMAC, ipv4CIDRs)
+	// Use the transport address for the Egress relevant external flows which is used for the Node connection.
+	if c.nodeConfig.NodeTransportIPv4Addr != nil && c.nodeConfig.PodIPv4CIDR != nil {
+		flows = c.externalFlows(c.nodeConfig.NodeTransportIPv4Addr.IP, *c.nodeConfig.PodIPv4CIDR, localGatewayMAC, ipv4CIDRs)
```

Comment:
The first argument `NodeIP` is used to identify traffic to local Node, I think both "NodeIP" and "NodeTransportIP" should be included.

Reply:
I didn't get your point clearly, do you mean we need two flows in function snatCommonFlows, one matches the NodeIP, and the other matches node transport IP? https://github.com/antrea-io/antrea/blob/main/pkg/agent/openflow/pipeline.go#L2109

Reply:
Yes, if they are different. It is my understanding of the purpose of the flows, otherwise traffic to local Node IP will go to SNAT table, not sure the consequence. Maybe you or @jianjuns added it in the very beginning?

Reply:
Got it, updated